benchmarks/apps/babelstream/babelstream.py

```diff
@@ -0,0 +1,261 @@
+# slightly different on the different letters, see above. (Compute Capability 70)
+# J-type nodes each have two NVIDIA Tesla P100s. (Compute Capability 60)
+            ################################################
+            # 'myriad': {
+            #     'Copy':  (25200, -0.05, 0.05, 'MB/s'),
+            #     'Scale': (16800, -0.05, 0.05, 'MB/s'),
+            #     'Add':   (18500, -0.05, 0.05, 'MB/s'),
+            #     'Triad': (18800, -0.05, 0.05, 'MB/s')
+            # },
+            #################################################
+            'tursa': {
+                'Copy':  (0, None, 600000, 'MB/s'),
+                'Scale': (0, None, 600000, 'MB/s'),
+                'Add':   (0, None, 600000, 'MB/s'),
+                'Triad': (0, None, 600000, 'MB/s')
+            },
+        }
+#--------------------------------------------------
+#-- OpenCL
+#--------------------------------------------------
+@rfm.simple_test
+class OCLBenchmark_CPU(BabelstreamBenchmarkBase):
+    valid_systems = ['-gpu']
+    tags = {"ocl"}
+    executable = "ocl-stream"
+
+@rfm.simple_test
+class OCLBenchmark_GPU(BabelstreamBenchmarkBase):
+    valid_systems = ['+gpu']
+    tags = {"ocl"}
+    executable = "ocl-stream"
+    num_gpus_per_node = 1
+#--------------------------------------------------
+#-- Kokkos
+#--------------------------------------------------
+@rfm.simple_test
+class KOKKOSBenchmark_CPU(BabelstreamBenchmarkBase):
+    valid_systems = ['-gpu']
+    tags = {"kokkos"}
+    executable = "kokkos-stream"
+
+@rfm.simple_test
+class KOKKOSBenchmark_GPU(BabelstreamBenchmarkBase):
+    valid_systems = ['+gpu']
+    tags = {"kokkos"}
+    executable = "kokkos-stream"
+    num_gpus_per_node = 1
+#--------------------------------------------------
+#-- TBB
+#--------------------------------------------------
+@rfm.simple_test
+class TBBBenchmark(BabelstreamBenchmarkBase):
+    valid_systems = ['-gpu']
+    tags = {"tbb"}
+    executable = "tbb-stream"
+    def __init__(self):
+        self.reference = {
+            'isambard-macs:cascadelake': { # https://en.wikichip.org/wiki/intel/xeon_gold/6230https://en.wikichip.org/wiki/intel/xeon_gold/6230 (SINGLE)
+                'Copy':  (0, None, 231860, 'MBytes/sec'),
+                'Mul': (0, None, 231860, 'MBytes/sec'),
+                'Add':   (0, None, 231860, 'MBytes/sec'),
+                'Triad': (0, None, 231860, 'MBytes/sec'),
+                'Dot': (0, None, 231860, 'MBytes/sec')
+            },
+
+        }
+#--------------------------------------------------
+#-- Thrust
+#--------------------------------------------------
+@rfm.simple_test
+class THRUSTBenchmark_NVIDIA(BabelstreamBenchmarkBase):
+    valid_systems = ['+gpu +cuda']
+    tags = {"thrust"}
+    executable = "thrust-stream"
+    num_gpus_per_node = 1
+
+@rfm.simple_test
+class THRUSTBenchmark_AMD(BabelstreamBenchmarkBase):
+    valid_systems = ['+gpu']
```

Comment:
> 

Thanks for mentioning I will add it to `_AMD` ones but is it better to use `+rocm` or `-cuda` ?

Reply:
My understanding is that this requires AMD GPUs, so an explicit `+rocm` would be better, to avoid confusion when we'll ever have Intel (or other vendors) GPUs